wp-admin/js/widgets/text-widgets.js

```diff
@@ -0,0 +1,186 @@
+	'use strict';
+
+	var component = {};
+
+	/**
+	 * Text widget control.
+	 *
+	 * @class TextWidgetControl
+	 * @constructor
+	 * @abstract
+	 */
+	component.TextWidgetControl = Backbone.View.extend({
+
+		/**
+		 * View events.
+		 *
+		 * @type {Object}
+		 */
+		events: {},
+
+		/**
+		 * Initialize.
+		 *
+		 * @param {Object}         options - Options.
+		 * @param {Backbone.Model} options.model - Model.
+		 * @param {jQuery}         options.el - Control container element.
+		 * @returns {void}
+		 */
+		initialize: function initialize( options ) {
+			var control = this;
+
+			Backbone.View.prototype.initialize.call( control, options );
+
+			if ( ! control.el ) {
+				throw new Error( 'Missing options.el' );
+			}
+		},
+
+		/**
+		 * Render template.
+		 *
+		 * @returns {void}
+		 */
+		render: function render() {
+			var control = this, changeDebounceDelay = 1000, id, editor, textarea;
+			textarea = control.$el.find( 'textarea:first' );
+			id = textarea.attr( 'id' );
+
+			// Destroy any existing editor so that it can be re-initialized after a widget-updated event.
+			if ( tinymce.get( id ) ) {
+				delete tinymce.editors[ id ];
+				tinymce.remove( '#' + id );
+			}
+
+			wp.editor.initialize( id, {
+				tinymce: {
+					wpautop: true
+				},
+				quicktags: true
+			} );
+
+			editor = window.tinymce.get( id );
+			editor.on( 'change', _.debounce( function() {
+				editor.save();
+				textarea.trigger( 'change' );
+			}, changeDebounceDelay ) );
+		}
+	});
+
+	/**
+	 * Mapping of widget ID to instances of TextWidgetControl subclasses.
+	 *
+	 * @type {Object.<string, wp.textWidgets.TextWidgetControl>}
+	 */
+	component.widgetControls = {};
+
+	/**
+	 * Handle widget being added or initialized for the first time at the widget-added event.
+	 *
+	 * @param {jQuery.Event} event - Event.
+	 * @param {jQuery}       widgetContainer - Widget container element.
+	 * @returns {void}
+	 */
+	component.handleWidgetAdded = function handleWidgetAdded( event, widgetContainer ) {
+		var widgetContent, widgetForm, idBase, widgetControl, widgetId;
+		widgetForm = widgetContainer.find( '> .widget-inside > .form, > .widget-inside > form' ); // Note: '.form' appears in the customizer, whereas 'form' on the widgets admin screen.
+		widgetContent = widgetForm.find( '> .widget-content' );
+
+		idBase = widgetForm.find( '> .id_base' ).val();
+		if ( 'text' !== idBase ) {
+			return;
+		}
+
+		widgetId = widgetForm.find( '> .widget-id' ).val();
+
+		// @todo Obtain the Save/Apply button and detatch the TinyMCE container element upon clicking, to then restore upon widgetUpdated.
+
+		// Prevent initializing already-added widgets.
+		if ( component.widgetControls[ widgetId ] ) {
+			return;
+		}
+
+		widgetControl = new component.TextWidgetControl({
+			el: widgetContent
+		});
+
+		// Add delay as without this the iframe may not get contenteditable.
+		_.delay( function() {
+			widgetControl.render();
+		}, 100 ); // eslint-disable-line no-magic-numbers
```

Comment:
According to my tests the wp.editor.initialize function call takes ±100ms to run - both on first initialisation, and on subsequent TinyMCE initialisations just as well.
This seems to match the delay you experienced here with it requiring 100ms of 'visibility' before someone can interact with them.

Correlation != causality... but my guess is that this is the browser's way to protect the computer resources by delaying any scripting that belongs to iframe's that are not visible.
I would recommend hiding it with something else than display: none (maybe position: absolute; left: -999em; instead) ... and see whether the browser behaves differently and allows the scripting to take place even before the user interacts with it and the iframe turns into becoming visible.